### PR TITLE
feat: implement test renumbering

### LINKED ISSUE #22: feat: add test renumbering
## Use Case

Sometimes tests need to be added in the middle of a test file, for consistency with previous tests when a rule chantes, or maybe to add negative tests.

But then you end up with all the tests after it with bad numbering :(

A nice feature will be to add a renumber on the test title so all tests are named properly.

⚠️ One caveat might be if there is a test that is excluded in a test file and we are renumbering it. This can be warned before renumbering by checking `.ftw.yaml,$HOME/.ftw.yaml` files to see if there is any _testoverride_ defined that might need to be changed, and warn the user.

### PR DESCRIPTION
**Pull request by theseion**
Fixes #22.

### CODE REVIEW

File: cmd/renumber_tests_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	suite.Run(t, new(renumberTestsTestSuite))
+}
+
+func (s *renumberTestsTestSuite) TestRenumberTests() {
+	contents := `---
+	meta:
+	  enabled: true
+	  name: 123456.yaml
+	tests:
+	  - test_title: bapedibupi
```
Comment (fzipi) [inline cmd/renumber_tests_test.go:53]: 👀 🏁

File: cmd/renumber_tests_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	  - test_title: "pine apple"
+		desc: "test 2"`
+	expected := `---
+	meta:
+	  enabled: true
+	  name: 123456.yaml
+	tests:
+	  - test_title: 123456 - 1
+		desc: "test 1"
+	  - test_title: 123456 - 2
```
Comment (fzipi) [inline cmd/renumber_tests_test.go:64]: We don't leave space between the test number and the dash, right? For this case it should be `123456-1` not `123456 - 1`, right?

File: regex/processors/context.go [hunk 3]
```diff
@@ -7,16 +7,19 @@ import (
...
-	singleChainOffset  bool
+	rootDirectory         string
+	rulesDirectory        string
+	utilDirectory         string
+	dataFilesDirectory    string
+	includeFilesDirectory string
+	includeFiles          string
+	singleRuleID          int
+	singleChainOffset     bool
 }
```
Comment (fzipi) [inline regex/processors/context.go:23]: Do we need two different contexts?

**Review by fzipi — CHANGES_REQUESTED** [review]
Left comments.

File: cmd/renumber_tests_test.go (hunk 2 above)
Comment (theseion) [reply cmd/renumber_tests_test.go:64]: Oh ok. I can change that.

File: regex/processors/context.go (hunk 3 above)
Comment (theseion) [reply regex/processors/context.go:23]: I need the directories for other things, but not the rest. That's why I moved them to a separate context. Not too happy with how it turned out... I was hoping that Karel would clean that up later 😜 .

Comment (fzipi) [reply regex/processors/context.go:23]: But it is confusing still. Why importing `"github.com/theseion/crs-toolchain/v2/context"` then? Maybe you want to extend it?

**Review by fzipi — APPROVED** [review]

File: regex/processors/context.go (hunk 3 above)
Comment (theseion) [reply regex/processors/context.go]: Do you like this version better? I do like it a little bit better.

**Review by fzipi — APPROVED** [review]
Perfect
